### PR TITLE
ENT-3276: Merge syspurpose with subscription-manager

### PR DESCRIPTION
**Pull request by jirihnidek**
* There is no syspurpose CLI tool anymore
* All functionality of syspurpose CLI is part of
  subscription-manager
* Moved few files from ./syspurpose/src/syspurpose package
  to ./src/syspurpose (files and utils modules)
* Moved some syspurpose unit tests to subscription-manager
  unit tests directory
* Fixed Makefile and .spec file. The syspurpose RPM is not
  build anymore
* Fixed .spec file to mark several directories as part of
  subscription-manager RPM
* Deleted most of .po files. Most of translation was related
  to syspurpose CLI
* Updated REDAME.md a little

### CODE REVIEW

**Comment by jirihnidek** [comment]
I need to fix failed tests and I need to obsolete syspurpose rpms in spec file too. I will fix it tomorrow.

**Comment by wottop** [comment]
I had this error when I executed tito build -i --rpm --test:

Traceback (most recent call last):
  File "/usr/share/rhsm-plugins/container_content.py", line 73, in <module>
    main()
  File "/usr/share/rhsm-plugins/container_content.py", line 57, in main
    from subscription_manager import injectioninit
  File "/usr/lib64/python3.8/site-packages/subscription_manager/injectioninit.py", line 20, in <module>
    from subscription_manager.cache import ProductStatusCache, EntitlementStatusCache, \
  File "/usr/lib64/python3.8/site-packages/subscription_manager/cache.py", line 38, in <module>
    from syspurpose.files import post_process_received_data
ImportError: bad magic number in 'syspurpose': b'\x03\xf3\r\n'

I ran this on a system that had syspurpose installed previously.

File: subscription-manager.spec [hunk 1]
```diff
@@ -1112,6 +1105,13 @@ find %{buildroot} -name \*.py* -exec touch -r %{SOURCE0} '{}' \;
 %{python_sitearch}/rhsmlib/services/__pycache__
 %endif
 
+# syspurpose
+%dir %{python_sitearch}/syspurpose
```
Comment (wottop) [inline subscription-manager.spec:1109]: What is the purpose of this section? Not saying it is wrong, just want to know why it is needed.
I see that most of the lines around it were removed below, but these were retained.

**Comment by jirihnidek** [comment]
> I had this error when I executed tito build -i --rpm --test:
> 
> Running scriptlet: subscription-manager-plugin-container-1.29.7-1.git 9/29
> Traceback (most recent call last):
> File "/usr/share/rhsm-plugins/container_content.py", line 73, in
> main()
> File "/usr/share/rhsm-plugins/container_content.py", line 57, in main
> from subscription_manager import injectioninit
> File "/usr/lib64/python3.8/site-packages/subscription_manager/injectioninit.py", line 20, in
> from subscription_manager.cache import ProductStatusCache, EntitlementStatusCache,
> File "/usr/lib64/python3.8/site-packages/subscription_manager/cache.py", line 38, in
> from syspurpose.files import post_process_received_data
> ImportError: bad magic number in 'syspurpose': b'\x03\xf3\r\n'
> 
> I ran this on a system that had syspurpose installed previously.

It looks like a conflict with old code installed in the system and new code, because method `post_process_received_data` is not available in module `subscription_manager.syspurposelib` is available only in `syspurpose.files`.

I wonder why tito tries to run this code during building of rpm? BTW: I don't have `/usr/share/rhsm-plugins/container_content.py` installed in the system. I guess that it is some container plugin, but why is it triggered during building of rpm?

File: subscription-manager.spec (hunk 1 above)
Comment (jirihnidek) [reply subscription-manager.spec:1109]: Good question. Some files were moved from `syspurpose/src/syspurpose` to the `src/syspurpose`. So these files are installed to the system (`/usr/lib64/python3.6/`) like `rhsmlib` is installed to the system. So the purpose of this section is to mark these installed files as part of subscription-manager RPM.

**Comment by wottop** [comment]
As long as the issue I am seeing is not going to happen when the package is updated via RPM install, then we should be good.

How do we advance our development environments? Once this is merged, a tito build will not allow us to run.

**Comment by jirihnidek** [comment]
> As long as the issue I am seeing is not going to happen when the package is updated via RPM install, then we should be good.
> 
> How do we advance our development environments? Once this is merged, a tito build will not allow us to run.

OK, I will try to test this issue with conflicting code more.

This PR is blocked by this PR: #2446

**Comment by jirihnidek** [comment]
@wottop I updated the PR, can you please re-try bulding RPMs? I tested it with installed container plugin and building/installing worked as expected. Tito build works as expected on Jenkins too.

**Comment by wottop** [comment]
When I tito build from master then do a tito build from your branch, I still get this:

http://pastebin.test.redhat.com/940724

**Comment by cnsnyder** [comment]
I also run into this issue when building this branch on f33: 
```
FileNotFoundError: [Errno 2] No such file or directory: "/tmp/tito/rpmbuild-subscription-managero9l5mt_s/SOURCES/subscription-manager-git-19.4fb7de3/warning: line 418: It's not recommended to have unversioned Obsoletes: Obsoletes: python3-syspurpose"
```

**Comment by jirihnidek** [comment]
> I also run into this issue when building this branch on f33:
> 
> ```
> FileNotFoundError: [Errno 2] No such file or directory: "/tmp/tito/rpmbuild-subscription-managero9l5mt_s/SOURCES/subscription-manager-git-19.4fb7de3/warning: line 418: It's not recommended to have unversioned Obsoletes: Obsoletes: python3-syspurpose"
> ```

Can we address this issue, when we provide `python3-syspurpose` for RHEL8, Fedora and we will provide sub-man for RHEL9? I think that we do want to have unversioned Obsoletes, because we want to remove any previous version of `python3-syspurpose`.

**Comment by jirihnidek** [comment]
> When I tito build from master then do a tito build from your branch, I still get this:
> 
> http://pastebin.test.redhat.com/940724

Hmm, interesting ... I think that I installed the rpms from master, then I tried install rpms fro this branch and I didn't see this issue. I will retry.

**Comment by jirihnidek** [comment]
> > When I tito build from master then do a tito build from your branch, I still get this:
> > http://pastebin.test.redhat.com/940724
> 
> Hmm, interesting ... I think that I installed RPMs from master, then I tried install RPMs fro this branch and I didn't see this issue. I will retry.

@wottop I tried to reproduce it once again and I don't see any issue. I want to ask. Is PYTHONPATH exported, when you run `dnf install ./*.rpm`? If yes, what is content of this environment variable? There is conflict with old and new code, but I cannot find any reason, why we should see this issue in production. The RPM tries to run `/usr/share/rhsm-plugins/container_content.py` as post installation script and it runs new code from subscription-manager that is already installed in the system.

**Comment by wottop** [comment]
As long as @cnsnyder does not see any issues, then I am good to merge this.

**Comment by wottop** [comment]
It turns out that the 'bad magic number' error is related to a mismatch between .py and .pyc files. I cleaned up my .pyc files and the RPM install worked correctly. My issues are done.

**Pull request merged**